### ISSUE TITLE
About Input Component docs

### ISSUE THREAD

**Issue by DarkKris** [zeit-ui/react#54]
# About `docs`

The document states that the input component has type attributes, but there are errors in using type props when coding.

@zeit-ui/react version 0.0.1-beta.8

```typescript
<Input
    placeholder={"password"}
    value={password}
    onChange={(ev: any) => setPassword(ev.target.value)}
    type={"password"}   // have error in there
    clearable
/>
```

**Comment by ofekashery** [comment]
Good point.
@unix What is the reason for using ``& typeof defaultProps`` in each component  type?

**Comment by unix** [comment]
@DarkKris fixed at #56 , a new version will be released after merged.

@ofekashery 
This code is to solve two problems:

  1. Normaliaed use of `defaultProps` in function component.
  2. If  a property has an initial-value, override type to `optional`.

For example:

```
type Props = {
  name: string
}

const defaultProps = {
  name: 'initial value',
}
```

The `name` has an initial value, so types error should be ignore when the user dose not set `name` value. (There is no good solution to the default value of functional component in the React community, especially when type validation is required)
